Modules/Core/Transform/test/itkResampleInPlaceImageFilterTest.cxx

```diff
@@ -0,0 +1,145 @@
+  typename ImageType::DirectionType direction = outputImage->GetDirection();
+  typename ImageType::SpacingType   spacing = outputImage->GetSpacing();
+
+  typename ImageType::PointType     origin_d = baselineImage->GetOrigin();
+  typename ImageType::DirectionType direction_d = baselineImage->GetDirection();
+  typename ImageType::SpacingType   spacing_d = baselineImage->GetSpacing();
+
+  // Image info validation
+  bool result = false; // no difference by default
+  for (unsigned int i = 0; i < ImageType::ImageDimension; ++i)
+  {
+    result = (result || Validate(origin[i], origin_d[i], tol));
+    result = (result || Validate(spacing[i], spacing_d[i], tol));
+    for (unsigned int j = 0; j < ImageType::ImageDimension; ++j)
+    {
+      result = (result || Validate(direction(i, j), direction_d(i, j), tol));
+    }
+  }
+
+  // Voxel contents validation
+  using ImageConstIterator = itk::ImageRegionConstIterator<ImageType>;
+  ImageConstIterator it1(outputImage, outputImage->GetRequestedRegion());
+  ImageConstIterator it2(baselineImage, baselineImage->GetRequestedRegion());
+  it1.GoToBegin();
+  it2.GoToBegin();
+  while (!it1.IsAtEnd())
+  {
+    result = (result || Validate(it1.Get(), it2.Get(), tol));
+    ++it1;
+    ++it2;
+  }
+
+  return result;
+}
+
+int
+itkResampleInPlaceImageFilterTest(int argc, char * argv[])
+{
+  if (argc < 3)
+  {
+    std::cerr << "Wrong arguments!" << std::endl;
+    std::cerr << "Usage: ./" << argv[0] << " inputImage baselineImage outputImage" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 3;
+  using PixelType = short;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+  using ImagePointer = ImageType::Pointer;
+  using TransformType = itk::VersorRigid3DTransform<double>;
+  using FilterType = itk::ResampleInPlaceImageFilter<ImageType, ImageType>;
+
+  // Read in input test image
+  ImagePointer inputImage;
+  ITK_TRY_EXPECT_NO_EXCEPTION(inputImage = itk::ReadImage<ImageType>(argv[1]));
+
+  // Set up transforms
+  ImageType::PointType center;
+  center.Fill(0.0);
+  center[0] = 2.0; // In mm along X (RL-axis)
+  center[1] = 5.0; // In mm along Y (AP-axis)
+  center[2] = 7.0; // In mm along Z (IS-axis)
+
+  itk::Vector<double, Dimension> translation;
+  translation.Fill(0.);
+  translation[0] = 10.0; // In mm along X (RL-axis)
+  translation[1] = 15.0; // In mm along Y (AP-axis)
+  translation[2] = 20.0; // In mm along Z (IS-axis)
+
+  itk::Vector<double, Dimension> rotationAxis;
+  rotationAxis[0] = 0.1;
+  rotationAxis[1] = 0.2;
+  rotationAxis[2] = 0.7;
+
+  double rotationAngle = .5; // Radians
+
+  TransformType::Pointer transform = TransformType::New(); // Identity by default
+  transform->SetCenter(center);
+  transform->Translate(translation);
+  transform->SetRotation(rotationAxis, rotationAngle);
+
+  // Set up the resample filter
+  FilterType::Pointer filter = FilterType::New();
+  filter->SetInputImage(inputImage);
```

Comment:
Can the corresponding Get method be exercised and be compared with the expected output?